### PR TITLE
Added Multisig and updated related files

### PR DESCRIPTION
**Pull request by harveenSingh**
- Added multisig-v0.3.0
- Added the related workflows
- Updated and added related files

### CODE REVIEW

File: .github/workflows/unitpacker-go.yml [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+  config:
+    runs-on: ubuntu-latest
+    outputs:
+      config: ${{ steps.set-config.outputs.config }}
+    steps:
+      - uses: actions/checkout@v3
+      - id: set-config
+        run: |
+          if [ -n "${ACT+x}" ]; then
+            ./compile-unitpacker-config.sh gaiad-v99.0.1
```
Comment (greg-szabo) [inline .github/workflows/unitpacker-go.yml]: Shouldn't this be something like `multisig-v...` and a different script for testing?
```suggestion
            ./compile-unitpacker-config-tools.sh multisig-v99.0.1
```

File: .github/workflows/unitpacker-go.yml [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+      - name: Print binary version
+        run: ~/go/bin/${{ matrix.tools.name }} --version
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: ${{ github.run_id }}.${{ matrix.os }}
+          if-no-files-found: error
+          retention-days: 3
+          path: ~/go/bin/${{ matrix.tools.name }}
```
Comment (greg-szabo) [inline .github/workflows/unitpacker-go.yml]: For future proofing, I would enable uploading Windows-compatible binaries here. (Windows requires the full name to upload, which is `multisig.exe` in our case`.)

By the way, did you try to build this for Windows? I think it should be doable. Although without the network binaries, it's less useful. :(

```suggestion
          path: ~/go/bin/${{ matrix.tools.name }}*
```

File: .github/workflows/unitpacker-go.yml [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+          p12-file-base64: ${{ secrets.MAC_APP_CERTIFICATE }}
+          p12-password: ${{ secrets.MAC_APP_CERTIFICATE_PASSPHRASE }}
+
+      - name: Sign macOS binary
+        if: runner.os == 'macOS'
+        run: codesign --timestamp -o runtime --sign "${{ secrets.MAC_APP_CERTIFICATE_CN }}" -v ./${{ matrix.tools.name }}
+
+      - name: Verify macOS signature
+        if: runner.os == 'macOS'
+        run: | ##
```
Comment (greg-szabo) [inline .github/workflows/unitpacker-go.yml]: Are the comments necessary there? Just asking...

```suggestion
        run: |
```

File: .github/workflows/unitpacker-go.yml [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+          codesign -v ./${{ matrix.tools.name }}
+          codesign --display --verbose=4 ./${{ matrix.tools.name }} 2>&1 | grep Authority
```
Comment (greg-szabo) [inline .github/workflows/unitpacker-go.yml]: Was the output too verbose so you grepped for Authority? (I guess you use this for debugging, right?)

I'll leave the option here to not grep, but it's up to you how it makes more sense for you.

```suggestion
          codesign --display --verbose=4 ./${{ matrix.tools.name }}
```

File: .github/workflows/unitpacker-go.yml [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+
+      - name: Staple package
+        if: runner.os == 'macOS'
+        run: |
+          xcrun stapler staple ${{ matrix.tools.name }}_${{ matrix.tools.ref }}_mac.pkg
+          xcrun stapler validate ${{ matrix.tools.name }}_${{ matrix.tools.ref }}_mac.pkg
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: ${{ github.run_id }}.macos-12.signed
```
Comment (greg-szabo) [inline .github/workflows/unitpacker-go.yml]: Something's fishy here: `mac-12` will break when GitHub Actions upgrades to macos-13, macos-14, etc. Also, we have to upload any Linux-generated files too.

Is this what you had in mind?
```suggestion
          name: ${{ github.run_id }}.${{ matrix.os }}.signed
```

File: compile-unitpacker-config-tools.sh [hunk 6]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+# GitHub Actions script to gather configuration from the local config.json file
+set -eux
+
+# Find tool configuration from config.json - Output: TOOL_CONFIG 
+TOOL_NAME="$(echo "$1" | sed 's/^\(.*\)-v.*$/\1/')"
+TOOL_CONFIG="$(jq '.tools | map(select(.name == "'"$TOOL_NAME"'"))[0]' config.json)"
+
+# Get the 'os' and 'tool' for unitpacker.
+CONFIG="$(jq -s '{os: .[0].os} + {tools: .[0].tools}' config.json "${TOOL_NAME}_config.json" | tr -d '\n')"
```
Comment (greg-szabo) [inline compile-unitpacker-config-tools.sh]: I don't understand what `"${TOOL_NAME}_config.json"` does here. There's no such file (ex: multisig_config.json doesn't exist). `jq` doesn't write a file like that but even if it did, it wouldn't be used for anything.

```suggestion
CONFIG="$(jq -s '{os: .[0].os} + {tools: .[0].tools}' config.json | tr -d '\n')"
```

File: config.json [hunk 7]
```diff
@@ -39,6 +39,11 @@
       "name": "ixod",
       "build": "v2",
       "version": "v0.18.1"
+    },
+    {
+      "name": "multisig",
+      "build": "v0.3.0",
```
Comment (greg-szabo) [inline config.json]: It's not a good idea to match the build version with the binary version. The build version's purpose is to differentiate among different GitHub Action builds of the binary. Multiple build versions can point to the same binary version.

The binary version is unique to the binary.

If you find a bug in the GitHub actions now and you need to rebuild the `multisig v0.3.0` binary, what will the build version be? `v0.3.1`? (You're not supposed to remove tags from the repo regularly.)

I would strongly suggest a `v1` build for multisig with the binary `v0.3.0` packaged.

**Review by greg-szabo — CHANGES_REQUESTED** [review]
Good work, minor things mentioned/suggested. Feel free to either accept the suggestions and close the thread or comment and ask for clarification / submit a counterpoint so we can discuss things we disagree on.

File: .github/workflows/unitpacker-go.yml (hunk 2 above)
Comment (harveenSingh) [reply .github/workflows/unitpacker-go.yml]: Agree with this. Updated the code.

File: .github/workflows/unitpacker-go.yml (hunk 3 above)
Comment (harveenSingh) [reply .github/workflows/unitpacker-go.yml]: Not necessary. Was a typo. Thanks

File: .github/workflows/unitpacker-go.yml (hunk 4 above)
Comment (harveenSingh) [reply .github/workflows/unitpacker-go.yml]: That is correct. I was using this to debug. Removed it now.

File: .github/workflows/unitpacker-go.yml (hunk 5 above)
Comment (harveenSingh) [reply .github/workflows/unitpacker-go.yml]: This was missed. It is updated now.

File: compile-unitpacker-config-tools.sh (hunk 6 above)
Comment (harveenSingh) [reply compile-unitpacker-config-tools.sh]: You are right. I was testing something but this is not needed. It is update now.

File: config.json (hunk 7 above)
Comment (harveenSingh) [reply config.json]: Thanks for this, I agree. It is now updated to v1

File: .github/workflows/unitpacker-go.yml (hunk 1 above)
Comment (harveenSingh) [reply .github/workflows/unitpacker-go.yml]: @greg-szabo For this, do we even need worry if github actions is being run locally? The variable "$GITHUB_REF_NAME" is the same for local github actions. So we could remove the if else condition and have it like this

`run: ./compile-unitpacker-config-tools.sh "$GITHUB_REF_NAME"`

Comment (greg-szabo) [reply .github/workflows/unitpacker-go.yml]: It's up to you. The condition was put in place for troubleshooting locally. (Running GitHub Actions locally. You don't get a GIT_REF then.) If you're not troubleshooting locally, you don't need it.

**Pull request merged**